### DISCUSSION TITLE
Allow sharing datamodel classes between client and server

### DISCUSSION THREAD

**Discussion by asierpn** [Azure/azure-mobile-apps#929] [Ideas]
We have started to share datamodel classes between server and client in our project, but to make it work I had to change this method in the ServiceSerializer class because the Version column on the client side is now of type byte[] like on the server side:

```
        public static string GetVersion(JObject instance)
        {
            foreach (JProperty property in instance.Properties())
            {
                if (string.Equals(property.Name, SystemProperties.JsonVersionProperty, StringComparison.OrdinalIgnoreCase))
                {
                    return JsonConvert.SerializeObject(instance.Value<byte[]>(property.Name)).Replace("\"", "");
                }
            }
            return null;
        }
```

It would be great if the new version of the library allowed this, so that the same classes could be reused between server and client.

**Comment by hansmbakker** [comment]
Development has moved to https://github.com/CommunityToolkit/Datasync (see https://github.com/Azure/azure-mobile-apps/discussions/912) - I suggest to move this item there

**Comment by asierpn** [comment]
Moved to https://github.com/CommunityToolkit/Datasync/issues/47
